rspace/src/main/scala/coop/rchain/rspace/nextgenrspace/SpaceMatcher.scala

```diff
@@ -0,0 +1,137 @@
+package coop.rchain.rspace.nextgenrspace
+
+import cats.effect.Sync
+import cats.implicits._
+import cats.syntax._
+import coop.rchain.catscontrib._
+import coop.rchain.rspace._
+import coop.rchain.rspace.history.Branch
+import coop.rchain.rspace.internal._
+import coop.rchain.rspace.trace.Log
+import coop.rchain.shared.SyncVarOps
+
+import scala.annotation.tailrec
+import scala.concurrent.SyncVar
+
+/** The interface for RSpace
+  *
+  * @tparam C a type representing a channel
+  * @tparam P a type representing a pattern
+  * @tparam A a type representing an arbitrary piece of data
+  * @tparam R a type representing a match result
+  * @tparam K a type representing a continuation
+  */
+private[rspace] trait SpaceMatcher[F[_], C, P, A, R, K] extends ISpace[F, C, P, A, R, K] {
```

Comment:
github sees this as a "new file" , better for maintainability if that was considered "moved and edit"

Reply:
It wasn't moved, we need to preserve the original version for now